tools/monorepo-utils/src/changefile/index.ts

```diff
@@ -133,6 +128,22 @@ const program = new Command( 'changefile' )
 					}
 				} );
 
+				if ( touchedProjectsRequiringChangelog.length === 0 ) {
+					Logger.notice( 'No projects require a changelog' );
+					process.exit( 0 );
+				}
+
+				// When a devRepoPath is provided, assume that the dependencies are already installed.
+				if ( ! devRepoPath ) {
+					Logger.notice(
+						`Installing dependencies in ${ tmpRepoPath }`
+					);
+					execSync( 'pnpm install', {
```

Comment:
Wow so we call the composer changelogger directly on each package. reee. this is so inefficient. (It makes the action super slow esp because we have to run all those postinstalls.)

Its fine for now obviously, but like.. can the changelogger be installed at the top level of the repo and just run for each package path in future or something? Otherwise I guess I'd really like to see us make that port of the changelogger to JS.

Reply:
>Wow so we call the composer changelogger directly on each package. Is that why we do this? I see the need of course but, this is rough 😂

yes, its rough. And the reason why I put this step until the last possible. I tried doing just `composer install`, but there are some JS dependencies too for some packages.

>can the changelogger be installed at the top level of the repo and just run for each package path in future or something?

Thats a good idea. I will look into it as a follow up